common/plugins/eu.esdihumboldt.hale.common.instance/src/eu/esdihumboldt/hale/common/instance/model/impl/MultiInstanceCollection.java

```diff
@@ -194,6 +194,7 @@ public Instance next() {
 						currentIterator.close();
 					currentCollection = hasNextIndex;
 					currentIterator = collections.get(currentCollection).iterator();
+					currentIterator.hasNext(); // force to initialize currentIterator.hasNext flag 
```

Comment:
You are right. With the change in https://github.com/stempler/hale/pull/3/files/499d2980a4f7f6bbaaae954abacf5027e0a1966a#diff-5ddc8e3e73548a88d2325686f278149f
this change is not required anymore.